plugins/CoreAdminHome/Commands/ConfigGet.php

```diff
@@ -0,0 +1,343 @@
+        $this->addOption('format', 'f', InputOption::VALUE_OPTIONAL, 'Format the output as ' . json_encode(self::OUTPUT_FORMATS) . '; Default is ' . self::OUTPUT_FORMAT_DEFAULT, self::OUTPUT_FORMAT_DEFAULT);
+
+        $this->setHelp("This command can be used to get a INI config setting or a whole section of settings on a Piwik instance.
+
+You can get config values per the two sections below, where:
+- [Section] is the name of the section, e.g. database or General.
+- [config_setting_name] the name of the setting, e.g. username.
+
+(1) By settings options --section=[Section] and --key=[config_setting_name].  The option --section is required. Examples:
+#Return all settings in this section.
+$ ./console %command.name% --section=database
+#Return only this setting.
+$ ./console %command.name% --section=database --key=username
+
+OR
+
+(2) By using a command argument in the format [Section].[config_setting_name]. Examples:
+#Return all settings in this section.
+$ ./console %command.name% 'database'
+#Return all settings in this array; square brackets are optional.
+$ ./console %command.name% 'PluginsInstalled.PluginsInstalled'
+$ ./console %command.name% 'PluginsInstalled.PluginsInstalled[]'
+#Return only this setting.
+$ ./console %command.name% 'database.username'
+
+NOTES:
+- Section and key names are case-sensitive; so e.g. --section=Database fails but --section=database works.  Look in global.ini.php for the proper case.
+- If no matching section/setting is found, the string \"" . self::MSG_NOTHING_FOUND . "\" shows.
+- Else the output will be shown JSON-encoded.  You can use something like https://stedolan.github.io/jq to parse it.
+- If you ask for 'PluginsInstalled.PluginsInstalled[\"some_array_item\"]', it will ignore the array key (\"some_array_item\") and you will get back the whole array of values (e.g. all PluginsInstalled[] values).
+");
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        //Optionally could set this at runtime with: $output->getVerbosity() >= OutputInterface::VERBOSITY_VERBOSE.
+        $debug = false;
+
+        // Gather options, then discard ones with an empty value so we do not need to check for empty later.
+        $options = array_filter([
+            'section' => $input->getOption('section'),
+            'key' => $input->getOption('key'),
+            'format' => $input->getOption('format'),
+        ]);
+        $debug && fwrite(STDERR, PHP_EOL . __FUNCTION__ . '::Started with $options=' . (empty($options) ? '' : serialize($options)));
+
+        // If none specified, set default format.
+        if ( ! isset($options['format']) || empty($format = $options['format']) || ! in_array($format, self::OUTPUT_FORMATS, true)) {
+            $format = self::OUTPUT_FORMAT_DEFAULT;
+        }
+        // Since it is config and not input data, remove format from the options so we can just work with the section and format.
+        unset($options['format']);
```

Comment:
Maybe handling the format independently makes the code a bit cleaner
```suggestion
        // Gather options, then discard ones with an empty value so we do not need to check for empty later.
        $options = array_filter([
            'section' => $input->getOption('section'),
            'key' => $input->getOption('key'),
        ]);

        // If none specified, set default format.
        $format = $input->getOption('format');
        if ( empty($format) || ! in_array($format, self::OUTPUT_FORMATS, true)) {
            $format = self::OUTPUT_FORMAT_DEFAULT;
        }
```